request_adapter.go

```diff
@@ -1,46 +1,47 @@
+// Kiota also provides default implementations for these concepts.
+// Checkout:
+// - github.com/microsoft/kiota/authentication/go/azure
+// - github.com/microsoft/kiota/http/go/nethttp
+// - github.com/microsoft/kiota/serialization/go/json
+package abstractions
+
+import (
+	"context"
+	"github.com/microsoft/kiota-abstractions-go/store"
+
+	s "github.com/microsoft/kiota-abstractions-go/serialization"
+)
+
+// ErrorMappings is a mapping of status codes to error types factories.
+type ErrorMappings map[string]s.ParsableFactory
+
+// RequestAdapter is the service responsible for translating abstract RequestInformation into native HTTP requests.
+type RequestAdapter interface {
+	// Send executes the HTTP request specified by the given RequestInformation and returns the deserialized response model.
+	Send(context context.Context, requestInfo *RequestInformation, constructor s.ParsableFactory, errorMappings ErrorMappings) (s.Parsable, error)
+	// SendEnum executes the HTTP request specified by the given RequestInformation and returns the deserialized response model.
+	SendEnum(context context.Context, requestInfo *RequestInformation, parser s.EnumFactory, errorMappings ErrorMappings) (any, error)
+	// SendCollection executes the HTTP request specified by the given RequestInformation and returns the deserialized response model collection.
+	SendCollection(context context.Context, requestInfo *RequestInformation, constructor s.ParsableFactory, errorMappings ErrorMappings) ([]s.Parsable, error)
+	// SendEnumCollection executes the HTTP request specified by the given RequestInformation and returns the deserialized response model collection.
+	SendEnumCollection(context context.Context, requestInfo *RequestInformation, parser s.EnumFactory, errorMappings ErrorMappings) ([]any, error)
+	// SendPrimitive executes the HTTP request specified by the given RequestInformation and returns the deserialized primitive response model.
+	SendPrimitive(context context.Context, requestInfo *RequestInformation, typeName string, errorMappings ErrorMappings) (any, error)
+	// SendPrimitiveCollection executes the HTTP request specified by the given RequestInformation and returns the deserialized primitive response model collection.
+	SendPrimitiveCollection(context context.Context, requestInfo *RequestInformation, typeName string, errorMappings ErrorMappings) ([]any, error)
+	// SendNoContent executes the HTTP request specified by the given RequestInformation with no return content.
+	SendNoContent(context context.Context, requestInfo *RequestInformation, errorMappings ErrorMappings) error
+	// GetSerializationWriterFactory returns the serialization writer factory currently in use for the request adapter service.
+	GetSerializationWriterFactory() s.SerializationWriterFactory
+	// EnableBackingStore enables the backing store proxies for the SerializationWriters and ParseNodes in use.
+	EnableBackingStore(factory store.BackingStoreFactory)
```

Comment:
can you expand on why you're passing the factory here?

Reply:
This is largely to be consistent with the other implementations 

i.e  dotnet https://github.com/microsoft/kiota-abstractions-dotnet/blob/ac62342975c84d5a26abdfd90a2f98ea8644b1c4/src/ApiClientBuilder.cs#L60

and php https://github.com/microsoft/kiota-abstractions-php/blob/24aba77dde3a1df97c15bc76a2850149af58efda/src/ApiClientBuilder.php#L23

Reply:
It also allows users to pass a different iplemenetation of a backing store

Reply:
I believe you're mixing concerns here, the dotnet link is to enable the backing store on the request adapter. The PHP one is just to register an serialization provider.
For former doesn't accept a parameter, the latter does.
Let me know if I need to clarify further.

Reply:
@baywet, sorry this is the correct link https://github.com/microsoft/kiota-abstractions-dotnet/blob/ac62342975c84d5a26abdfd90a2f98ea8644b1c4/src/IRequestAdapter.cs#L22

Reply:
thanks for clarifying, resolving